### PR TITLE
Refactor how vendor scripts are added

### PR DESCRIPTION
**Pull request by ericf**
This revamps how vendor scripts are added to the site:

- The `Intl` polyfill is now loaded via the Polyfill service
- No more trying to build the FormatJS libs into one file
- FormatJS integration libs are only served on the pages that need them

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

File: Brocfile.js [hunk 1]
```diff
@@ -47,54 +46,43 @@ var vendor = new Funnel('public/vendor/', {
...
+// Create a roll-up file of locale data for the locales the app supports for
+// each integration lib.
 var formatjsLocaleData = mergeTrees([
     'dust-intl',
     'handlebars-intl',
     'react-intl'
 ].map(function (integration) {
-    return new Funnel(node_modules, {
+    return concatTree(new Funnel(node_modules, {
         srcDir : integration + '/dist/locale-data',
```
Comment (ericf) [inline Brocfile.js]: This might not be the place where Ember Intl's locale data is…

File: shared/components/splash-example.jsx [hunk 2]
```diff
@@ -3,12 +3,9 @@
 import LocaleSelect from './locale-select';
 
 var CSSTransitionGroup = React.addons.CSSTransitionGroup;
-var IntlMixin          = ReactIntl.IntlMixin;
-var FormattedMessage   = ReactIntl.FormattedMessage;
 
 export default React.createClass({
     displayName: 'SplashExample',
-    mixins     : [ReactIntl.IntlMixin],
```
Comment (ericf) [inline shared/components/splash-example.jsx]: This really didn't need to be using the ReactIntl Mixin.

**Comment by caridy** [comment]
LGTM

**Pull request merged**